packages/lmi/tests/test_rate_limiter.py

```diff
@@ -293,3 +295,22 @@ async def test_embedding_rate_limits(
         )
     else:
         assert estimated_tokens_per_second > 0
+
+
+@pytest.mark.asyncio
+async def test_try_acquire():
+    TEST_RATE_CONFIG = {
+        ("get", "test"): RateLimitItemPerSecond(30, 4),
+    }
+
+    with patch.object(GLOBAL_LIMITER, "rate_config", TEST_RATE_CONFIG):
+        # We can acquire 30 in less than 10 seconds
+        for _ in range(30):
+            await GLOBAL_LIMITER.try_acquire(("get", "test"))
+
+        # But if we try for one more we hit the limit
+        with pytest.raises(TimeoutError):
+            await GLOBAL_LIMITER.try_acquire(("get", "test"), acquire_timeout=2)
+
+        # Then we pass an impossible limit, but with a timeour high enough to succeed
```

Comment:
```suggestion
        # Then we pass an impossible limit, but with a timeout high enough to succeed
```

Re-reading this, I just noticed a typo

Reply:
For fun I made https://github.com/codespell-project/codespell/pull/3656 for this, lol